tests/integration/records/test_records_api.py

```diff
@@ -92,3 +93,16 @@ def test_create_does_not_save_zombie_identifiers_if_record_creation_fails(isolat
 
     assert not record_identifier
     assert not persistent_identifier
+
+
+def test_citations_count_equals_zero(isolated_app):
+    record = get_db_record('lit', 1497201)
+    assert record.get_citations_count() == 0L
+
+
+def test_citations_count_non_zero(isolated_app):
+    record_1 = get_db_record('lit', 1477591)
+    assert record_1.get_citations_count() == 1L
```

Comment:
you don't know who's populating the citation counts for demo records: your code or https://github.com/inspirehep/inspire-next/blob/4b538c9529715e7ebdcb9ee3a6eb92f4ec778410/scripts/recreate_records#L43 ? If you remove the old code in the `migrator` module then you're sure it's yours (as an alternative, you can also insert a pair of citing/cited records in the DB and check the count is correct, but the old code will have to die anyway).

Reply:
I know that data in `InspireHep.get_citations_count()` are not taken from citation_count field in ES which is created by the old code but they are coming from index in DB `ix_records_metadata_json_referenced_records` and this index is created by me.